chainntnfs/btcdnotify/btcd.go

```diff
@@ -772,6 +817,20 @@ func (b *BtcdNotifier) RegisterSpendNtfn(outpoint *wire.OutPoint,
 				return nil, err
 			}
 		}
+	} else {
+		// Otherwise, if the output is still unspent, then we'll note
+		// the current height as its height hint.
+		_, currentHeight, err := b.chainConn.GetBestBlock()
```

Comment:
not sure if this is safe. Imagine we query for the UTXO, we get it back, but before we commit the spend hint a new block comes in that increase the height by one, potentially including a spend.

Reply:
Fixed.